data/src/main/java/org/apache/iceberg/data/FileAccessFactoryRegistry.java

```diff
@@ -0,0 +1,269 @@
+    return writeBuilderFor(format, objectModelName, outputFile, FileContent.DATA);
+  }
+
+  /**
+   * Creates a writer builder for generating a {@link DeleteFile} with equality deletes.
+   *
+   * <p>The returned builder produces a writer that accepts records defined by the specified object
+   * model and persists them using the given file format. These specialized writers collect equality
+   * delete records that identify rows to be deleted based on equality conditions, producing a
+   * {@link DeleteFile} that can be used for table operations.
+   *
+   * @param format the file format used for writing
+   * @param objectModelName name of the object model defining the input format
+   * @param outputFile destination for the written data
+   * @param <E> type of the engine-specific schema expected by the writer
+   * @return a configured delete write builder for creating an {@link EqualityDeleteWriter}
+   */
+  public static <E> EqualityDeleteWriteBuilder<?, E> equalityDeleteWriteBuilder(
+      FileFormat format, String objectModelName, EncryptedOutputFile outputFile) {
+    return writeBuilderFor(format, objectModelName, outputFile, FileContent.EQUALITY_DELETES);
+  }
+
+  /**
+   * Creates a writer builder for generating a {@link DeleteFile} with position-based deletes.
+   *
+   * <p>The returned builder produces a writer that accepts records defined by the specified object
+   * model and persists them using the given file format. These specialized writers collect {@link
+   * PositionDelete} records that identify rows to be deleted by file path and position, producing a
+   * {@link DeleteFile} that can be used for table operations.
+   *
+   * @param format the file format used for writing
+   * @param objectModelName name of the object model defining the input format
+   * @param outputFile destination for the written data
+   * @param <E> type of the engine-specific schema expected by the writer
+   * @return a configured delete write builder for creating a {@link PositionDeleteWriter}
+   */
+  public static <E> PositionDeleteWriteBuilder<?, E> positionDeleteWriteBuilder(
+      FileFormat format, String objectModelName, EncryptedOutputFile outputFile) {
+    return writeBuilderFor(format, objectModelName, outputFile, FileContent.POSITION_DELETES);
+  }
+
+  @SuppressWarnings("unchecked")
+  private static <B extends WriteBuilder<B, E>, E>
+      ContentFileWriteBuilderImpl<?, ?, E> writeBuilderFor(
+          FileFormat format,
+          String objectModelName,
+          EncryptedOutputFile outputFile,
+          FileContent content) {
+    return new ContentFileWriteBuilderImpl<>(
+        ((FileAccessFactory<E>) FILE_ACCESS_FACTORIES.get(new Key(format, objectModelName)))
+            .<B>writeBuilder(outputFile.encryptingOutputFile(), content),
+        outputFile.encryptingOutputFile().location(),
+        format);
+  }
+
+  /** Key used to identify readers and writers in the {@link FileAccessFactoryRegistry}. */
```

Comment:
Do we need a special class here? We already have Pair I think which basically does this and isn't private?

Reply:
Moved to using Pair.
Originally I have chickened out using the Pair because of the schema cache, but after double checking, I have realized that it is not used in our case.
Thanks for the suggestion!